doc/contributing.rst

```diff
@@ -1,221 +1,760 @@
+    $ git remote -v  # show existing remote addresses
+    $ git remote set-url origin git@github.com:$GITHUB_USERNAME/mne-python.git
+    $ git remote set-url upstream git@github.com:mne-tools/mne-python.git
 
-Checking and building documentation
------------------------------------
 
-All changes to the codebase must be properly documented.
-To ensure that documentation is rendered correctly, the best bet is to
-follow the existing examples for class and function docstrings,
-and examples and tutorials.
+MNE-Python coding conventions
+=============================
 
-Our documentation (including docstring in code) uses ReStructuredText format,
-see `Sphinx documentation`_ to learn more about editing them. Our code
-follows the `NumPy docstring standard`_.
+General requirements
+^^^^^^^^^^^^^^^^^^^^
 
-Documentation is automatically built remotely during pull requests. If
-you want to also test documentation locally, you will need to install (using
-``conda`` or ``pip`` as preferred):
+All new functionality must have test coverage
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For example, a new :class:`mne.Evoked` method in :file:`mne/evoked.py` should
+have a corresponding test in :file:`mne/tests/test_evoked.py`.
+
+
+All new functionality must be documented
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This includes thorough docstring descriptions for all public API changes, as
+well as how-to examples or longer tutorials for major contributions. Docstrings
+for private functions may be more sparse, but should not be omitted.
+
+
+New API elements should be added to the master reference
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Classes, functions, methods, and attributes cannot be cross-referenced unless
+they are included in the :doc:`python_reference`
+(:file:`doc/python_reference.rst`).
+
+
+Describe your changes in the changelog
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Include in your changeset a brief description of the change in the
+:doc:`changelog <whats_new>` (:file:`doc/whats_new.rst`; this can be skipped
+for very minor changes like correcting typos in the documentation). Note that
+there are sections of the changelog for each release, and separate subsections
+for bugfixes, new features, and changes to the public API. It is usually best
+to wait to add a line to the changelog until your PR is finalized, to avoid
+merge conflicts (since the changelog is updated with almost every PR).
+
+
+Test locally before opening pull requests (PRs)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MNE-Python uses `continuous integration`_ (CI) to ensure code quality and
+test across multiple installation targets. However, the CIs are often slower
+than testing locally, especially when other contributors also have open PRs
+(which is basically always the case). Therefore, do not rely on the CIs to
+catch bugs and style errors for you; run the tests locally instead before
```

Comment:
I would just make "run the tests" link to the section below, and get rid of the sentence that follows. (The link implies that they can get more info there if it's not immediately clear what they should do.)